pages/index.tsx

```diff
@@ -0,0 +1,245 @@
+//getting icon based on current weather
+export function getIconUrl(code: string): string {
+  return `http://openweathermap.org/img/wn/${code}.png`;
+}
+
+export default function Home() {
+  const API_KEY = process.env.NEXT_PUBLIC_API_KEY;
+  const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const [weatherData,setWeatherData] = useState<WeatherDetail>();
+  const [latitude,setLatitude] = useState(0);
+  const [longitude,setLongitude] = useState(0);
+  const [place,setPlace] = useState("Japan");
+  const [forecast,setForecast] = useState<WeatherData>();
+
+  const currentDate = new Date();
+  const year = currentDate.getFullYear();
+  const month = currentDate.getMonth() + 1;
+  const day = currentDate.getDate();
+  const todayDate= `${year}/${month}/${day}`;
+  
+  //page routing
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+  const handleOnChange = async (value:string) => {
+    setPlace(value);
+  };
+
+  useEffect(() => {
+    axios
+        .get(`https://api.openweathermap.org/geo/1.0/direct?q=${place}&limit=5&appid=${API_KEY}`) //getting coordinates based on place selected by user
+        .then((response) => {
+            const data = response.data;
+            setLatitude(data[0].lat);
+            setLongitude(data[0].lon)
+        });
+    axios
+        .get(`${BASE_URL}weather?lat=${latitude}&lon=${longitude}&appid=${API_KEY}&units=metric`) //getting current data
+        .then((response) => {
+            const data = response.data;
+            setWeatherData(data);
+        });
+    axios
+      .get(`${BASE_URL}forecast?lat=${latitude}&lon=${longitude}&appid=${API_KEY}&units=metric`) //getting next 5days data
+      .then((response) => {
+          const data = response.data;
+          setForecast(data);
+      });
+      const params = new URLSearchParams(searchParams); //updating url
+      if (place) {
+        params.set('query', place);
+      } else {
+        params.delete('query');
+      }
+      replace(`${pathname}?${params.toString()}`);
+    
+}, [place,latitude,longitude]); 
+
+function convertUnixTimeToDate(unixUtc: number): Date {
+  return new Date(unixUtc * 1000);
+}
+
+const uniqueDates = [
+  ...new Set(
+    forecast?.list.map(
+      (entry) => new Date(entry.dt * 1000).toISOString().split("T")[0]
+    )
+  )
+];
+// Filtering data to get the first entry after 6 AM for each unique date
+const firstDataForEachDate = uniqueDates.map((date) => {
+  return forecast?.list.find((entry) => {
+    const entryDate = new Date((entry.dt) * 1000).toISOString().split("T")[0];
+    const entryTime = new Date((entry.dt) * 1000).getHours();
+    return entryDate === date && entryTime >= 6;
+  });
+});
+
+//getting temperature array of next 5 days 
+const chart_temp = new Array<Chart>;
+firstDataForEachDate.map((d) => 
+  chart_temp.push({
+    temp:(Math.floor(d?.main.temp ?? 0)),
+    dates: (d ? format(parseISO(d.dt_txt), "dd.MM") : "")
+  }
+));
```

Comment:
Please try to separate the function to be another utils or new custom hook.

Reply:
Made the changes.